src/generic/Poly.jl

```diff
@@ -2743,16 +2743,38 @@ end
 #
 ###############################################################################
 
-function rand(rng::AbstractRNG, S::AbstractAlgebra.PolyRing, deg_range::UnitRange{Int}, v...)
+RandomExtensions.maketype(S::PolyRing, dr::UnitRange{Int}, _...) = elem_type(S)
+
+# define rand for make(S, deg_range, v)
+function rand(rng::AbstractRNG, sp::Random.SamplerTrivial{<:RandomExtensions.Make3{<:RingElement,<:PolyRing,UnitRange{Int}}})
```

Comment:
`Make3` is an alias for `Make` objects containing 3 elements; it's first type parameter if the type of produced objects, and the 3 subsequent type parameters correspond to the types of the 3 stored objects.